Upgrade clippy on CI

jtgeibel opened the pull request.

carols10cents commented on src/models/team.rs:
```diff
@@ -116,6 +116,7 @@ impl Team {
         // check that `team` is the `slug` in results, and grab its data
 
         // "sanitization"
+        #[cfg_attr(feature = "cargo-clippy", allow(trivially_copy_pass_by_ref))]
```
wdyt about, instead, changing this to `fn whitelist(c: char) -> bool` and line 126 to `if let Some(c) = org_name.chars().find(|c| whitelist(*c)) {`?

jtgeibel commented:
@carols10cents I incorporated your suggestion and rebased on master.

I also bumped to yesterday's nightly and addressed a new lint, as shown below:

```diff
diff --git a/src/bin/render-readmes.rs b/src/bin/render-readmes.rs
index 41f8fa42..62937e05 100644
--- a/src/bin/render-readmes.rs
+++ b/src/bin/render-readmes.rs
@@ -61,7 +61,7 @@ fn main() {
     let args: Args = Docopt::new(USAGE)
         .and_then(|d| d.deserialize())
         .unwrap_or_else(|e| e.exit());
-    let config: Config = Default::default();
+    let config = Config::default();
     let conn = cargo_registry::db::connect_now().unwrap();
 
     let start_time = Utc::now();
diff --git a/src/bin/server.rs b/src/bin/server.rs
index bd3e5ae4..ef263ce1 100644
--- a/src/bin/server.rs
+++ b/src/bin/server.rs
@@ -15,7 +15,7 @@ use std::sync::Arc;
 fn main() {
     // Initialize logging
     env_logger::init().unwrap();
-    let config: cargo_registry::Config = Default::default();
+    let config = cargo_registry::Config::default();
 
     // If there isn't a git checkout containing the crate index repo at the path specified
     // by `GIT_REPO_CHECKOUT`, delete that directory and clone the repo specified by `GIT_REPO_URL`
```

carols10cents commented:
LGTM!

bors: r+
